File: app/src/main/java/com/kode/utair/presentation/utils/AndroidResourceManager.java
package com.kode.utair.presentation.utils;

import android.content.Context;

import com.kode.utair.domain.commons.ResourceManager;

import javax.inject.Inject;

public class AndroidResourceManager implements ResourceManager {

    private Context context;

    @Inject
    public AndroidResourceManager(Context context) {
        this.context = context;
    }

    @Override
    public String getString(String resourceId) {
        return context.getResources().getString(getIdentifier(resourceId, "string"));
    }

    @Override
    public Integer getInteger(String resourceId) {
        return context.getResources().getInteger(getIdentifier(resourceId, "integer"));
    }

    private int getIdentifier (String name, String defType) {
       return context.getResources().getIdentifier(name, defType, context.getPackageName());
    }

}

File: core/src/main/java/com/utair/presentation/mvp/global/AndroidResourceManager.kt
package com.utair.presentation.mvp.global

import android.content.Context
import com.utair.domain.global.ResourceManager
import javax.inject.Inject

class AndroidResourceManager @Inject constructor(
        private val context: Context
) : ResourceManager {

    override fun getString(resourceId: Int, vararg args: Any): String {
        return context.resources.getString(resourceId, *args)
    }

    override fun getInteger(resourceId: Int): Int {
        return context.resources.getInteger(resourceId)
    }

}